addons/binding/org.openhab.binding.milight/src/main/java/org/openhab/binding/milight/internal/protocol/QueuedCategorizedSend.java

```diff
@@ -0,0 +1,182 @@
+        byte[] data;
+        int category;
+        boolean repeatable;
+
+        public QueueItem(byte[] data, int category, boolean repeatable) {
+            super();
+            this.data = data;
+            this.category = category;
+            this.repeatable = repeatable;
+        }
+    }
+
+    BlockingQueue<QueueItem> queue = new LinkedBlockingQueue<>(20);
+    protected final DatagramPacket packet;
+    protected final DatagramSocket datagramSocket;
+    private static final Logger logger = LoggerFactory.getLogger(QueuedCategorizedSend.class);
+    private int delay_between_commands = 100;
+    private int repeat_commands = 1;
+    private boolean willbeclosed = false;
+    private Thread thread;
+
+    public static final byte NO_CATEGORY = 0;
+
+    public QueuedCategorizedSend(InetAddress addr, int port) throws SocketException {
+        byte[] a = new byte[0];
+        packet = new DatagramPacket(a, a.length, addr, port);
+        datagramSocket = new DatagramSocket();
+        thread = new Thread(this);
+        thread.start();
+    }
+
+    public int getDelayBetweenCommands() {
+        return delay_between_commands;
+    }
+
+    public int getRepeatCommands() {
+        return repeat_commands;
+    }
+
+    public void setRepeatCommands(int repeat_commands) {
+        this.repeat_commands = repeat_commands;
+    }
+
+    public void setDelayBetweenCommands(int ms) {
+        delay_between_commands = ms;
+    }
+
+    /**
+     * The queue process
+     */
+    @Override
+    public void run() {
+        while (!willbeclosed) {
+            QueueItem item;
+            try {
+                // block/wait for another item
+                item = queue.take();
+            } catch (InterruptedException e) {
+                if (!willbeclosed) {
+                    logger.error("Queue take failed: " + e.getLocalizedMessage());
+                }
+                break;
+            }
+
+            if (item == null) {
+                continue;
+            }
+
+            packet.setData(item.data);
+            try {
+                int repeat_remaining = item.repeatable ? repeat_commands : 1;
+                repeat_remaining = Math.min(repeat_remaining, 1);
+
+                StringBuffer s = new StringBuffer();
+                for (int i = 0; i < item.data.length; ++i) {
+                    s.append(String.format("%02X ", item.data[i]));
+                }
+
+                while (repeat_remaining > 0) {
+                    datagramSocket.send(packet);
+                    --repeat_remaining;
+                    logger.debug("Sent packet '{}' to bridge {}", s.toString(), packet.getAddress().getHostAddress());
+                }
+
+            } catch (Exception e) {
+                logger.error("Failed to send Message to '{}': {}", packet.getAddress().getHostAddress(),
+                        e.getMessage());
+            }
+
+            try {
+                Thread.sleep(delay_between_commands);
```

Comment:
I recon the bridge can only handle a specific low amount of traffic in general, but we cannot control what other apps are sending whatsoever and the effort to realize bridge spanning delays for single bulbs is not worth it in my opinion